.github/workflows/release.yml

```diff
@@ -0,0 +1,94 @@
+name: release
+
+on:
+  push:
+    branches: [ stable ]
+  workflow_dispatch:
+
+jobs:
+
+  checks:
+    name: check releases
+    runs-on: ubuntu-latest
+    outputs:
+      current_version: ${{ steps.current_version.outputs.current_version }}
+      version_latest: ${{ steps.version_latest.outputs.version_latest }}
+      already_released: ${{ steps.version_latest.outputs.version_latest == steps.current_version.outputs.current_version }}
+      already_in_pypi: ${{ steps.check_in_pypi.outputs.pypi_versions != '' }}
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get current version
+        id: current_version
+        run: |
+          python -m pip install . --no-deps
+          out="$(pip show ${{ env.PKG_NAME }} | grep Version: | cut -d'' '' -f 2)"
+          echo "$out"
+          echo "::set-output name=current_version::$out"
+        shell: bash
+      - name: Get latest tag
+        id: version_latest
+        run: |
+          out="$(git describe --abbrev=0 --tags | cut -d'v' -f 2)"
+          echo "$out"
+          echo "::set-output name=version_latest::$out"
+        shell: bash
+      - name: check if already deployed to PyPI
+        id: check_in_pypi
+        # Note. other options:
+        #   - use 'pip install --no-deps PKG==VERSION' with current version
+        #   - use 'pip index versions PKG==VERSION'
+        #     (but it's a kind of experimental feature of pip >= 21.2)
+        run: |
+          python -m pip install --upgrade pip
+          out="$(pip install --use-deprecated=legacy-resolver ${{ env.PKG_NAME }}== 2>&1 \
+              | grep -E "Could not find .* ${{ steps.current_version.outputs.current_version }}(,|\))")"
+          echo "::set-output name=pypi_versions::$out"
+        shell: bash {0}  # to opt-out of default fail-fast behavior
+
+  release-github:
+    name: GitHub Release
+    runs-on: ubuntu-latest
+    needs: testjob1
+    if: github.event_name == 'push' && github.ref == 'refs/heads/stable'
+    steps:
+      - uses: actions/checkout@v2
+      - name: build dist
+        if: needs.testjob1.outputs.already_released == 'false'
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade build
+          python -m build
+      - uses: ncipollo/release-action@v1
```

Comment:
Please note, that as far as a I know, SICPA GitHib org has the following policy:
`Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub or verified in the GitHub Marketplace.`
So, I'm not sure that we can use that action.